launch_ros/launch_ros/actions/lifecycle_node.py

```diff
@@ -87,7 +89,7 @@ def __init__(
         # TODO(jacobperron): Remove default value and this check when deprecated API is removed
         if name is None:
             raise RuntimeError("'name' must not be None.'")
-        super().__init__(name=name, **kwargs)
+        super().__init__(name=name, namespace=namespace, **kwargs)
```

Comment:
If `namespace` is already passed in via `**kwargs`, then it seems like this changes the default from `None` to `''`, which makes the default value used by `LifecycleNode` different from the one used by `Node`. I think they should have the same default value if possible to avoid behavior differences between the two actions.

https://github.com/ros2/launch_ros/blob/e8894ce6155b516b6c5f4dda1c5b4cba76c3e725/launch_ros/launch_ros/actions/node.py#L73

Reply:
> If namespace is already passed in via **kwargs, then it seems like this changes the default from None to '', which makes the default value used by LifecycleNode different from the one used by Node

Yes, this is changing only the default.

`LifecycleNode` action must know the fully qualified node name at launch time, if not it can't interact with the lifecycle services.

The other option is:

> Make namespace argument required.

As I commented in the post description, that's my preferred solution.
The only disadvantage is that I will have to create multiple PRs.

Reply:
I agree with you @ivanpauno, but I don't quite grok why you need multiple PRs to enforce a  namespace is given. Why isn't updating [line 161](https://github.com/ros2/launch_ros/blob/e8894ce6155b516b6c5f4dda1c5b4cba76c3e725/launch_ros/launch_ros/actions/lifecycle_node.py#L161) enough?

Reply:
Every place where the namespace wasn't specified in a `LifecycleNode` action have to change:
https://github.com/ros2/demos/blob/bda392ed2e4ce4c9ad428935382e97502ba7c676/lifecycle/launch/lifecycle_demo.launch.py#L22-L23

PS: I think that's the only place luckily

Reply:
I see. If that's the sole location wouldn't it be better to just fix it?

Reply:
> LifecycleNode action must know the fully qualified node name at launch time, if not it can't interact with the lifecycle services.

Ah I see, `LifecycleNode` needs to know the namespace, so it makes sense for it to have different behavior from Node. +1 to `LifecycleNode` requiring the `namespace` parameter instead of defaulting to `/`

> I see. If that's the sole location wouldn't it be better to just fix it?

+1

Reply:
If I understand correctly, this alone will fix the test https://github.com/ros2/demos/pull/455

I still agree making LifecycleNode require namespace makes sense, but the test fix can go in first while that's being discussed.

Reply:
See https://github.com/ros2/demos/pull/456.

> I see. If that's the sole location wouldn't it be better to just fix it?

I wasn't sure what was the intended behavior the original author thought, and thus I took the easiest approach in the first iteration.
Maybe the intention was "use the root namespace implicitly if no namespace is specified".

If we want to backport some of the fixes in https://github.com/ros2/launch_ros/pull/153 and here, we will have to do something like that, as changing an optional argument to required is not acceptable in a backport.